### ISSUE TITLE
Import custom request file instead of copying it over to request.ts

### ISSUE THREAD

**Issue by lesha-co** [ferdikoomen/openapi-typescript-codegen#1438]
I have a following script in my package.json

```
 "openapi-codegen": "openapi -i 'https://whatever.com/whatever/swagger/?format=openapi' -o src/openapi --request src/OpenAPIRequestImplementation.ts",
```

What it does is puts generated classes inside `src/openapi` and copies `src/OpenAPIRequestImplementation.ts` to `src/openapi/core/request.ts`

That's fine I guess but implementation file has imported files such as 

```
import { ApiRequestOptions } from './ApiRequestOptions';
import { CancelablePromise } from './CancelablePromise';
import { OpenAPIConfig } from './OpenAPI';
```

These relative paths are fine for `core/request.ts` file but my code editor draws a lot of red squiggly lines under them because of course, these files do not exist in `src/` folder

**Isn't it better to just import file mentioned in `--request` switch instead of copying it into known location like that?:**

```
- import { request as __request } from '../core/request';
+ import { request as __request } from '../../OpenAPIRequestImplementation.ts';
```

I can't edit the request.ts file or put mine alongside because each time I run the codegen script (say, api has changed) the whole folder seem to get erased

**Comment by MaikBluemel** [comment]
I'm totally with you lesha-co. I have the same situation here in my project.
The generated import paths do not work anymore, because my custom request.ts has to use other imports than the generated one.

I don't know, how this feature is supposed to work. 
Maybe there is just a hint missing or I understand it wrong.

**Comment by muuhoffman** [comment]
an alternative interim solution might be to use absolute imports

**Comment by MaikBluemel** [comment]
Hi muuhoffman, 

I don't see, how this would help us. The problem is, that you can not take the custom_request.ts in the core-folder, because the generator will purge the complete directory and remove your custom_request.ts.
So if you place the custom_request.ts outside this folder, the folder will differ to the generated one.

My actual solution is, to put the file outside my src-folder and use the import-statements which will be correct, when the file is in the core-directory. But outside the src-folder, this file will be displayed as errornous, but it will not crash the compiling process.

Or maybe I am missing something?

**Comment by muuhoffman** [comment]
You have your custom_request.ts under your src/ directory

See how to setup typescript absolute imports here:
How to setup typescript absolute imports https://stackoverflow.com/questions/55916731/using-absolute-paths-in-typescript-for-imports

tsconfig.json
```
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "@/*": ["./src/*"],
    }
  }
}
```

Then all your imports can look like this: 
```
import { ApiError, CancelablePromise, OpenAPIConfig } from '@/openapi'
import { ApiRequestOptions } from '@/openapi/core/ApiRequestOptions'
import { ApiResult } from '@/openapi/core/ApiResult'
import { OnCancel } from '@/openapi/core/CancelablePromise'
```

**Comment by MaikBluemel** [comment]
Oh, now I see. Yes that works, thanks a lot.

For me I had to use the following config in tsconfig.json:

```
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "src/*": ["./src/*"],
    }
  }
}
```

Now my imports look like this:
```
import { ApiError, CancelablePromise, OpenAPIConfig } from 'src/openapi'
import { ApiRequestOptions } from 'src/openapi/core/ApiRequestOptions'
import { ApiResult } from 'src/openapi/core/ApiResult'
import { OnCancel } from 'src/openapi/core/CancelablePromise'
```

The only thing is that the @ did not work for me. But I am not very familiar with this feature.

Thank you very much.

**Comment by lesha-co** [comment]
I agree that it fixes the immediate problem, but copying a file is still a code smell IMO.

**Comment by MaikBluemel** [comment]
> I agree that it fixes the immediate problem, but copying a file is still a code smell IMO.

I agree, too. My custom_request.ts is not red anymore, that's fine. But I still have two request-files (my custom one and the one used by the client). Your proposal, lesha-co, would produce clean and comprehensible code without duplicates.

**Comment by defrex** [comment]
I generate multiple clients, so while the absolute imports solves importing shared functionality, I still can't reference the other generated code and use the request implementation for all the generated clients.

I see the internal-use implementation uses handlebars template for these files. (Here is [`fetch/request.hbs`](https://github.com/ferdikoomen/openapi-typescript-codegen/blob/master/src/templates/core/fetch/request.hbs), for example). As much as giving up on typechecking isn't ideal, exposing an override for the handlebars template would at least provide a clean path to a generic solution.
